features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -42,6 +43,12 @@ class Base64Serializer private constructor(
     ) {
         asyncImageProcessingCallback?.startProcessingImage()
 
+        val cachedBase64 = base64LruCache.get(drawable.hashCode())
```

Comment:
in order to avoid collisions you could better rely on `System.identityHashCode(drawable)` which always returns the memory address of that reference.

Reply:
as we discussed this will work for non `BitmapDrawables`  but for those you will have to use something like `System.identityHashCode(bitmap)`

Reply:
If I understand correctly, if we go by memory address then two separate drawables that resolve to the same bitmap would require separate cache entries? The contract for hashcode specifies that if the objects are equal then they get the same id.

Reply:
hmm, if you check the `hashCode` function into the `Object` class you will see that they delegate the same `identityHashCode` static function which is also called by `System.identityHashCode`. What I am suggesting here is to use that directly in order to avoid situations where users will provide their own implementation of a `Drawable` where they override the `hashCode` function and by doing that they could create collisions in your cache.

Reply:
you want to keep the `hashCode` here ?